src/components/TopBar/Annotations.js

```diff
@@ -48,6 +48,36 @@ export const Annotations = observer(({ store, annotationStore }) => {
     return () => document.removeEventListener('click', handleClick);
   }, []);
 
+  const commentsNumber = useMemo(() => {
+    let _unresolvedComments = 0;
+    let _comments = 0;
+
+    commentStore.comments.forEach(obj => {
+      _comments++;
+
+      if (!obj.isResolved) _unresolvedComments++;
+    });
+
+    return [_comments, _unresolvedComments];
+  }, [[...commentStore.comments]]);
+
+  useEffect(() => {
+    const [comments, unresolved] = commentsNumber;
+
+    commentStore.annotation.setUnresolvedCommentCount(unresolved);
+    commentStore.annotation.setCommentCount(comments);
+  }, [commentsNumber]);
```

Comment:
This will rerun the memo every single time as it is just creating a new array each pass. We should probably look at another solution or remove the memo as this is will be no different than:

```suggestion
  const commentsNumber = (() => {
    let _unresolvedComments = 0;
    let _comments = 0;

    commentStore.comments.forEach(obj => {
      _comments++;

      if (!obj.isResolved) _unresolvedComments++;
    });

    return [_comments, _unresolvedComments];
  })();

  useEffect(() => {
    const [comments, unresolved] = commentsNumber;

    commentStore.annotation.setUnresolvedCommentCount(unresolved);
    commentStore.annotation.setCommentCount(comments);
  }, [commentsNumber]);
```